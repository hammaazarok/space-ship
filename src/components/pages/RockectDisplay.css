.main {
  display: flex;
  gap: 2vw;
  margin: 2vw;
}

.container{
  border-top: solid 1px lightgray;
}

.main img {
  height: 14vw;
  width: 15vw;
}

.rocketDescription {
  text-align: left;
  line-height: 1;
  margin-left: 1rem;
}

/* .main button {
  background-color: rgb(17, 116, 255);
  padding: 0.5rem 1rem;
  outline: none;
  border: none;
  color: #fff;
  border-radius: 0.3rem;
} */


@media only screen and (max-width: 540px) {
  .main {
    flex-direction: column;
    align-items: center;
    justify-content: center;
  }

  .main > img {
    width: 80vw;
    height: auto;
    margin: 1rem auto;
  }

  .rocketImage {
    justify-self: center;
  }
}