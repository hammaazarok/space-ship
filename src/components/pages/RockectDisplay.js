/* eslint-disable camelcase */
// import { useSelector } from 'react-redux';

const RockectDisplay = () => {
  // const rockects = useSelector((state) => state.rockects);

  // const renderRockect = rockects.map((rockect) => (
  //   <div key={rockect.id}>
  //     <div>
  //       <img src={rockect.flickr_images} alt={rockect.rocket_name} />
  //     </div>
  //     <div>
  //       <h1>{rockect.rocket_name}</h1>
  //       <h2>{rockect.rocket_type}</h2>
  //       <p>
  //         {rockect.description}
  //       </p>
  //     </div>
  //   </div>
  // ));

  // return (
  //   <>{ renderRockect }</>
  // );
};

export default RockectDisplay;
