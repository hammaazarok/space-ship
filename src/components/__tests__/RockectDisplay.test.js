import React from 'react';
import Renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../../redux/configureStore';
import RockectDisplay from '../pages/RockectDisplay';

it('renders correctly', () => {
  const tree = Renderer
    .create(
      <React.StrictMode>
        <Provider store={store}>
          <BrowserRouter>
            <RockectDisplay
              key={0}
              id={0}
              reserved={false}
              name="rockect.rocket_name"
              description="rockect.description"
              img={[]}
              bookingHandler={() => {}}
            />
          </BrowserRouter>
        </Provider>
      </React.StrictMode>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
