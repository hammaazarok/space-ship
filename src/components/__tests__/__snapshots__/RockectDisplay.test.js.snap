// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="container"
>
  <div
    className="main"
  >
    <div
      className="rocketImage"
    >
      <img
        alt=""
        src={Array []}
      />
    </div>
    <div
      className="rocketDescription"
    >
      <h1>
        rockect.rocket_name
      </h1>
      <p>
        rockect.description
      </p>
      <button
        className="btn btn-reserve width-20vw btn btn-primary"
        disabled={false}
        id={0}
        onClick={[Function]}
        type="button"
      >
        Reserve Rocket
      </button>
    </div>
  </div>
</div>
`;
